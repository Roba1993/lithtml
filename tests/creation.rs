use indoc::indoc;
use insta::assert_snapshot;
use lithtml::{Dom, Node, Result};

#[test]
fn it_can_create_artefacts() -> Result<()> {
    let mut dom = Dom::new();
    dom.children.push(Node::Comment("Welcome to the test"));
    dom.children.push(Node::parse_json(indoc!(
        r#"{
          "name": "div",
          "variant": "normal",
          "children": [
            {
              "name": "h1",
              "variant": "normal",
              "children": [
                "Tjena världen!"
              ]
            },
            {
              "name": "p",
              "variant": "normal",
              "children": [
                "Tänkte bara informera om att Sverige är bättre än Finland i ishockey."
              ]
            }
          ]
        }"#
    ))?);
    dom.children.append(&mut Node::parse(
        r#"<div>Testing</div><p>Multiple elements from node</p>"#,
    )?);

    assert_snapshot!(dom);
    Ok(())
}
