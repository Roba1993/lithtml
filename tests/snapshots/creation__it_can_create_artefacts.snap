---
source: tests/creation.rs
assertion_line: 35
expression: dom
---
<!-- Welcome to the test -->
<div>
	<h1>Tjena världen!</h1>
	<p>
		Tänkte bara informera om att Sverige är bättre än Finland i ishockey.
	</p>
</div>
<div>Testing</div>
<p>Multiple elements from node</p>
